tests/acceptance/comparative_booster_study/dataset_definition.py

```diff
@@ -0,0 +1,701 @@
+import json
+from pathlib import Path
+
+from databuilder.query_language import Dataset, case, when
+
+from . import codelists, schema
```

Comment:
@evansd I've just tried running this (to dump the SQL for Graphnet).

```
❯ just exec python -m databuilder dump-dataset-sql --dataset-definition tests/acceptance/comparative_booster_study/dataset_definition.py databuilder.backends.tpp.TPPBackend
/home/benbc/bin/just _compile pyproject.toml requirements.prod.txt 
/home/benbc/bin/just _compile requirements.dev.in requirements.dev.txt 
$BIN/python -m databuilder dump-dataset-sql --dataset-definition tests/acceptance/comparative_booster_study/dataset_definition.py databuilder.backends.tpp.TPPBackend
2022-07-06 13:58:25 [info     ] Validating dataset for tests/acceptance/comparative_booster_study/dataset_definition.py [databuilder.main] 
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/benbc/src/opensafely-core/databuilder/databuilder/__main__.py", line 175, in <module>
    main()
  File "/home/benbc/src/opensafely-core/databuilder/databuilder/__main__.py", line 44, in main
    validate_dataset(
  File "/home/benbc/src/opensafely-core/databuilder/databuilder/main.py", line 52, in validate_dataset
    dataset_definition = load_definition(definition_file)
  File "/home/benbc/src/opensafely-core/databuilder/databuilder/main.py", line 81, in load_definition
    module = load_module(definition_file)
  File "/home/benbc/src/opensafely-core/databuilder/databuilder/main.py", line 102, in load_module
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "tests/acceptance/comparative_booster_study/dataset_definition.py", line 6, in <module>
    from . import codelists, schema
ImportError: attempted relative import with no known parent package
error: Recipe `exec` failed on line 137 with exit code 1
```

Relative imports don't play nicely with the way we currently import dataset definitions. Making it an absolute import fixes that problem -- I seem to remember that we add the enclosing directory to the sys path exactly to allow for this.

However I then get an error from `schema.py` where it's got a relative import that goes up a level, which I guess we just don't support for real studies and is only needed here because we're not using the real schema.

Reply:
I've been playing around with various options here and I've come to the conclusion that there's no _sane_ way of making a Python file that works as both a top-level module (i.e an argument to `python <file>.py`) and as module nested inside a package.

I think in this weird case where we want to run databuilder against a dataset definition nested deep inside the `tests` package, the simplest solution is a shim file which imports it:
```
echo 'from tests.acceptance.comparative_booster_study.dataset_definition import dataset' > shim.py
```

And then use that as the `--dataset-definition`:
```
python -m databuilder dump-dataset-sql --dataset-definition shim.py databuilder.backends.tpp.TPPBackend
```